### ISSUE TITLE
Designing and Implementing End Conditions

### ISSUE THREAD

**Issue by patrickkoopmans** [uchicago-cs/chiventure#262]
In this issue, we aim to create a way of checking and storing "end conditions". This will require the efforts of WDL to create a way to specify how a game should end, and game-state to store whatever information may be needed to check whether these "end conditions"have been satisfied.

**Comment by thoranderskow** [comment]
We have decided that for this sprint it is probably good to have implemented being in a certain room as an end condition if that is possible. Supposedly ending the game when you have a certain item in inventory is a backlog issue.

**Comment by patrickkoopmans** [comment]
The game struct currently has the field "room_t *curr_room;" which should make this easy to implement. We can easily store "room_t* final_room;" in the game struct as well, as long as you are able to assign the final_room pointer the right pointer in a third pass through all the rooms. 

We could potentially add a check at the beginning of our move_room() function in game.h to see if the room being travelled to is the final_room. @thoranderskow let me know what you think!

**Comment by thoranderskow** [comment]
That sounds fine, I believe that we will be able to assign the final_room pointer in a game_struct. If adding a check to your move_room() function is what works best for you, then I say it sounds like a good way to implement ending.

**Comment by thoranderskow** [comment]
a room_t final_room in the game_t struct works as well.

**Comment by patrickkoopmans** [comment]
@thoranderskow I have implemented the changes and updated the header file documentation of move_room(). The function is found within our src directory in game.c and our include directory in game.h. I am working on branch game-state/end_condition. Once you confirm that these changes work for you and nothing was communicated incorrectly, I will update our README and pull request to master so you can use this for the prototype.

**Comment by thoranderskow** [comment]
These changes sound good, and everything seems to have been addressed correctly. We will be able to load the pointer to the final room into the game_t struct.

**Comment by patrickkoopmans** [comment]
I will write an add_final_room_to_game function as well to make this easier

**Comment by patrickkoopmans** [comment]
This issue is complete now that all comments on pull request #305 are resolved. 

In this issue, teams game-state and WDL worked together to design and implement some way of ending a game. WDL decided that the best way to go about this (given the time frame) was to include a final_room room_t pointer to the game struct which would store the "final room" of the game. The idea behind this end condition is that once the player enters the room (i.e. room_t* curr_room points to the final room) the game will end. WDL has accordingly added this feature to their files, and I have made changes to a number of game state files in order to accommodate this feature:

1. Added a room_t* final_room pointer within the game struct that WDL will use to load the game into
2. Implemented the function add_final_room_to_game() which initializes the final room and checks that the room exists within the list of rooms in the game
3. Edited the implementation of move_room() to return FINAL_ROOM (a macro) when the player moves to the final room
4. Added a number of other macros such as GAME_NULL and ROOM_NULL to game_state_common.h in order to reduce the cognitive load of the move_room() function return values

**Todo:** 
A more complete implementation of end conditions would involve using the condition structs within game_action.h in order to create a list of conditions that must be satisfied before the game ends. This method of end condition is far more complex, and is only possible due to very recent developments within the game action module. A separate issue (#347) has been backlogged to keep track of this issue. 

Thank you @thoranderskow for your help with this!
